### ISSUE TITLE
Client fails to reconnect on connection error

### ISSUE THREAD

**Issue by mszabo-wikia** [KittehOrg/KittehIRCClientLib#219]
The IRC client fails to reconnect to the server if its connection is closed, even when the `ClientConnectionEndedEvent` indicates it will try to reconnect (and a handler explicitly instructs it to do so).

I'm attaching the following minimal, self-contained test snippet that should assist in reproducing the issue:
```java
import net.engio.mbassy.listener.Handler;
import org.kitteh.irc.client.library.Client;
import org.kitteh.irc.client.library.element.Channel;
import org.kitteh.irc.client.library.event.client.ClientConnectionEndedEvent;
import org.kitteh.irc.client.library.event.client.ClientConnectionEstablishedEvent;

import java.util.stream.Collectors;

public class IrcClientDemo {

  private final String targetChannel;

  private IrcClientDemo(String targetChannel) {
    this.targetChannel = targetChannel;
  }

  public static void main(String[] args) {
    String targetChannel = "#my-channel";
    IrcClientDemo ircClientDemo = new IrcClientDemo(targetChannel);

    Client client = Client.builder()
        .nick("jkowalski")
        .serverHost("my-irc-server")
        .serverPort(6697)
        .exceptionListener(Throwable::printStackTrace)
        .secure(false)
        .build();

    client.getEventManager().registerEventListener(ircClientDemo);

    client.connect();
  }

  @Handler
  public void handleSuccessfulConnection(ClientConnectionEstablishedEvent event) {
    System.out.println("IRC connection successful");
    event.getClient().addChannel(targetChannel);
  }

  @Handler
  public void handleConnectionError(ClientConnectionEndedEvent event) {
    event.getCause().ifPresent(Throwable::printStackTrace);

    if (event.canAttemptReconnect()) {
      System.out.println("disconnected from server, trying to reconnect");

      System.out.println("target channels: " + event.getClient().getChannels().stream().map(Channel::getName).collect(Collectors.joining(",")));

      event.setAttemptReconnect(true);
      event.setReconnectionDelay(0);
    } else {
      System.out.println("disconnected from server, cannot reconnect");
    }
  }
}
```

**Steps to reproduce**
* run the snippet, wait for the client to join your channel
* terminate the client's connection by issuing a `/kill` IRC command
* **DEFECT:** The client indicates it will reconnect to the server and channel (based on `ClientConnectionClosedEvent` context), but reconnection never comes.

**Expected Behavior**
* client should reconnect, as indicated by event info

**Comment by mbax** [comment]
A week later than I hoped, but I have fixed it. Tested against QUIT and KILL.
https://github.com/KittehOrg/KittehIRCClientLib/commit/a62fc51cbc16863e994447e3935dfd135c543864

This will be available as 4.0.1 once maven central gets around to it.

**Comment by mszabo-wikia** [comment]
Thanks! :)
